### PR TITLE
Remove gui1 codebase

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Removes the old GUI1 code base and reduces the Rust code footprint by removing unused code.

### Important Notes

Updates build scripts and reformats part of the codebase with the autoformatter. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by mwu-tow — CHANGES_REQUESTED** [review]
While not run as part of PR checks, CI still builds old IDE for releases.
Find `UploadIde` and remove it. Similarly, `DeployGui` job should go — it deploys the old GUI to the cloud.

**Review by somebody1234 — APPROVED** [review]
approving `ide-desktop` changes. most of them seem to be pretty minor.

**Comment by somebody1234** [comment]
also we should consider removing the local dashboard (`app/ide-desktop/lib/content/`) in favor of embedding the `cloud.enso.org` dashboard (`app/ide-desktop/lib/dashboard/`) directly into the electron app (`app/ide-desktop/lib/content/`) by editing the build scripts (`bundle.ts`)

**Review by JaroslavTulach — APPROVED** [review]

**Comment by farmaazon** [comment]
> also we should consider removing the local dashboard (`app/ide-desktop/lib/content/`) in favor of embedding the `cloud.enso.org` dashboard (`app/ide-desktop/lib/dashboard/`) directly into the electron app (`app/ide-desktop/lib/content/`) by editing the build scripts (`bundle.ts`)

Let's make it a separate issue.

File: app/gui2/config.yaml [hunk 1]
```diff
@@ -1,30 +1,30 @@
 # The name of an object created in the global window scope. This object will contain functions
 # to control the IDE, like the `main` function, and also the configuration object.
-windowAppScopeName: "enso"
+windowAppScopeName: 'enso'
```
Comment (farmaazon) [inline app/gui2/config.yaml:3]: This was meant to be shared configuration between rust and js projects; but now we have only js, and gui2 does not even use this file. We should reconsider if we want to keep this file (not in this PR, but perhaps continuation of the task).

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -436,12 +273,6 @@ jobs:
     name: Publish release (linux, x86_64)
     needs:
       - enso-build-ci-gen-draft-release-linux-x86_64
-      - enso-build-ci-gen-job-deploy-gui-linux-x86_64
-      - enso-build-ci-gen-job-deploy-runtime-linux-x86_64
-      - enso-build-ci-gen-upload-ide-linux-x86_64
-      - enso-build-ci-gen-upload-ide-macos-aarch64
-      - enso-build-ci-gen-upload-ide-macos-x86_64
-      - enso-build-ci-gen-upload-ide-windows-x86_64
       - enso-build-ci-gen-upload-ide2-linux-x86_64
```
Comment (farmaazon) [inline .github/workflows/release.yml:276]: Let's rename ide2/gui2 to ide/gui everywhere.

File: lib/rust/enso-pack/Cargo.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "ensogl-pack"
+name = "enso-pack"
```
Comment (farmaazon) [inline lib/rust/enso-pack/Cargo.toml:2]: I think we don't need this crate anymore. The rust remnants are compiled by gui2's package.json script directly calling wasm-pack.

File: Cargo.toml [hunk 4]
```diff
@@ -7,33 +7,26 @@ resolver = "2"
...
+    "lib/rust/parser/doc-parser",
+    "lib/rust/parser/src/syntax/tree/visitor",
+    "lib/rust/parser/jni",
+    "lib/rust/parser/generate-java",
+    "lib/rust/parser/schema",
+    "lib/rust/parser/debug",
+    "lib/rust/enso-pack",
+    "tools/language-server/logstat",
+    "tools/language-server/wstest",
 ]
```
Comment (farmaazon) [inline Cargo.toml:27]: Technically, every path dependency of a workspace member is also a workspace member IIRC. So this list could be reduced just to `lib/rust*` perhaps? `build` directory could also have only one, main package listed.

File: Cargo.toml [hunk 5]
```diff
@@ -7,33 +7,26 @@ resolver = "2"
...
 # The default memebers are those we want to check and test by default.
-default-members = ["app/gui", "lib/rust/*"]
+default-members = ["lib/rust/*"]
```
Comment (farmaazon) [inline Cargo.toml]: The default members were specified to not run integration tests by default. Let's remove this now.

File: build/ci_utils/src/actions/artifacts.rs [hunk 6]
```diff
@@ -12,7 +12,6 @@ use serde::de::DeserializeOwned;
 use tempfile::tempdir;
 
 
-
```
Comment (farmaazon) [inline build/ci_utils/src/actions/artifacts.rs:15]: Why do we remove this line?

File: lib/rust/launcher-shims/README.md [hunk 7]
```diff
@@ -1,58 +0,0 @@
-# Launcher Shims
```
Comment (farmaazon) [inline lib/rust/launcher-shims/README.md]: Are you sure Engine don't need this crate anymore?

File: lib/rust/parser/src/macros/expand.rs [hunk 8]
```diff
@@ -5,6 +5,9 @@ use crate::prelude::*;
 
 use crate::syntax;
 
+use std::collections::HashMap;
+use std::collections::HashSet;
```
Comment (farmaazon) [inline lib/rust/parser/src/macros/expand.rs:9]: Arent these already imported by prelude?

File: lib/rust/parser/src/lib.rs [hunk 9]
```diff
@@ -240,15 +238,15 @@ fn expression_to_statement(mut tree: syntax::Tree<'_>) -> syntax::Tree<'_> {
...
-                    }
+                            // If the LHS is a type, this is a (destructuring) assignment.
+                            let lhs = expression_to_pattern(mem::take(lhs));
+                            tree.variant = Box::new(Variant::Assignment(Assignment {
+                                pattern: lhs,
+                                equals: mem::take(opr),
+                                expr: mem::take(rhs),
+                            }));
+                            return tree;
+                        }
```
Comment (farmaazon) [inline lib/rust/parser/src/lib.rs:249]: Bad formatting

File: lib/rust/shapely/macros/src/derive_entry_point.rs [hunk 10]
```diff
@@ -7,8 +7,6 @@ use crate::prelude::*;
 // ===================
 
 fn crate_name_to_base_name(name: &str) -> String {
-    let name = name.replace("debug-scene-", "");
-    let name = name.replace("ensogl-example-", "");
```
Comment (farmaazon) [inline lib/rust/shapely/macros/src/derive_entry_point.rs:11]: We no longer have special entry points, so I think this entire module could be removed.

File: lib/rust/types/src/unit.rs [hunk 11]
```diff
@@ -1,874 +1,940 @@
...
-//! to implement many impls because the type `Unit` would be defined in an external crate.
-//!
-//! ## Macro Naming Pattern
-//! Many macros defined in this module use a specific naming, for example
-//! `impl_UNIT_x_FIELD_to_UNIT`. You should interpret it as a macro defining a transformation
-//! similar to `fn<Unit,Field>(unit:Unit, field:Field) -> Unit`, where `Field` is a specific field
-//! of the `Unit` (in most cases units have single field).
+//! [`Duration`] together, or divide one [`Duration`] by a number or another [`Duration`] and get
+//! [`Duration`] or a number, respectfully. You are allowed to define any combination of operators
+//! and rules of how the result inference should be performed.
```
Comment (farmaazon) [inline lib/rust/types/src/unit.rs:8]: So, I guess we don't keep the deprecated module, but replace it with unit2, and just the GH does not understand it?

**Review by farmaazon — APPROVED** [review]
I think it's ok to merge (if it does not break CI), but it needs follow-up PR (or issue) which will remove enso(gl)-pack.

File: .github/workflows/release.yml (hunk 2 above)
Comment (farmaazon) [reply .github/workflows/release.yml:276]: Or, after thinking about it a bit, I think we may leave old names - renaming may take a bit of time, and we want to [move and rename gui2 later](8484)

File: app/gui2/config.yaml (hunk 1 above)
Comment (MichaelMauderer) [reply app/gui2/config.yaml:3]: It's still used in some parts of the JS code base, so it should be removed once those are refactored.

File: lib/rust/enso-pack/Cargo.toml (hunk 3 above)
Comment (MichaelMauderer) [reply lib/rust/enso-pack/Cargo.toml:2]: I'll double-check and prepare a follow-up PR.

File: Cargo.toml (hunk 4 above)
Comment (MichaelMauderer) [reply Cargo.toml:27]: Changing it as suggested does cause build issues, do I'd opt for leaving it as is.

File: build/ci_utils/src/actions/artifacts.rs (hunk 6 above)
Comment (MichaelMauderer) [reply build/ci_utils/src/actions/artifacts.rs:15]: This was introduced by running `./run fmt` so I'd opt to leave it in.

File: lib/rust/launcher-shims/README.md (hunk 7 above)
Comment (MichaelMauderer) [reply lib/rust/launcher-shims/README.md]: Double-checked with Radek and will restore it. It's not really used currently, but also not deprecated.

File: lib/rust/parser/src/macros/expand.rs (hunk 8 above)
Comment (MichaelMauderer) [reply lib/rust/parser/src/macros/expand.rs:9]: To reduce the size of Prelude (and eventually remove it) that was removed.

File: lib/rust/types/src/unit.rs (hunk 11 above)
Comment (MichaelMauderer) [reply lib/rust/types/src/unit.rs:8]: Correct, there are just a few bits left that are used in the parser.

File: lib/rust/parser/src/macros/expand.rs (hunk 8 above)
Comment (farmaazon) [reply lib/rust/parser/src/macros/expand.rs:9]: Hmm, and do we aim for prelude _removal_? I assumed it should be just slimmed down. And the common imports are one of the basic role of it.

**Review by mwu-tow — APPROVED** [review]

**Pull request merged**